paywall/src/unlockTypes.ts

```diff
@@ -51,3 +51,61 @@ export interface Network {
 export interface ChainExplorerURLBuilders {
   [site: string]: (address: string) => string
 }
+
+export interface Lock {
+  address: string
+  transaction: string
+}
+
+export interface Key {
+  lock: string
+  id: string
+  owner: string
+  status: string
+  expiration: number
+  transactions?: Transaction[]
+}
+
+export interface Locks {
+  [address: string]: Lock
+}
+
+export interface PaywallCallToAction {
+  default: string
+  expired: string
+  pending: string
+  confirmed: string
+}
+
+// This interface describes an individual paywall's config
+export interface PaywallConfig {
+  icon: string
+  callToAction: PaywallCallToAction
+  locks: Locks
+}
+
+export enum KeyStatus {
+  NONE = 'none',
+  CONFIRMING = 'confirming',
+  CONFIRMED = 'confirmed',
+  EXPIRED = 'expired',
+  VALID = 'valid',
+}
+
+export interface UILock {
+  name: string
+  address: string
+  keyPrice: string
+  expirationDuration: number
+  key: UIKey
+}
+
+export interface UILocks {
+  [address: string]: UILock
+}
+
+export interface UIKey {
+  expiration: number
+  transactions: Transaction[]
+  status: string
```

Comment:
can add `confirmations: number` here too if that makes sense

Reply:
key will also have a lock member, which is the lock address. Since it's not needed, you can include it for completeness or not